src/update-feeds-loops/submit-transactions.ts

```diff
@@ -196,12 +298,25 @@ export const submitTransactions = async (
   api3ServerV1: Api3ServerV1,
   updatableDataFeeds: UpdatableDataFeed[],
   blockNumber: number
-) =>
-  Promise.all(
-    updatableDataFeeds.map(async (dataFeed) =>
-      submitTransaction(chainId, providerName, provider, api3ServerV1, dataFeed, blockNumber)
-    )
-  );
+) => {
+  const {
+    config: { walletDerivationScheme },
+  } = getState();
+  if (walletDerivationScheme.type === 'fixed') {
+    if (isEmpty(updatableDataFeeds)) {
+      return [];
+    }
+    return Array.from({ length: updatableDataFeeds.length }).fill(
+      await submitBatchTransaction(chainId, providerName, provider, api3ServerV1, updatableDataFeeds, blockNumber)
```

Comment:
This should be called only once (for all updatable feeds).

Reply:
It is. I'm just filling the array with the result of this single call to not mess with the [`successCount`](https://github.com/api3dao/airseeker-v2/blob/single-sponsor-wallet/src/update-feeds-loops/update-feeds-loops.ts#L303) since it is based on data feed updates.

Reply:
I see. The code was a bit more cryptic to me and I misread it.